homeassistant/components/zwave_js/config_flow.py

```diff
@@ -1241,11 +1243,15 @@ async def async_step_restore_failed(
         """Restore failed."""
         if user_input is not None:
             return await self.async_step_restore_nvm()
+        assert self.backup_filepath is not None
+        assert self.backup_data is not None
 
         return self.async_show_form(
             step_id="restore_failed",
             description_placeholders={
-                "file_path": str(self.backup_filepath),
+                "file_path": self.backup_filepath,
+                "file_url": f"data:application/octet-stream;base64,{base64.b64encode(self.backup_data).decode('ascii')}",
+                "file_name": os.path.basename(self.backup_filepath),
```

Comment:
I'd make `self.backup_filepath` a `Path` instance from the beginning and then we can use `Path.name` here and copy to a full string path where that is needed.